datahub_sap_hana/ingestion.py

```diff
@@ -1,40 +1,177 @@
+    """Creates a datasource for the lineage of tables from a SAP HANA database. 
+    It contains the connection to the db using sqlachemy to get the table metadata to construct the downstream-upstream lineage. 
+    Returns an iterable work unit that gets emitted to Datahub."""
+
+    config: HanaConfig
+
     def __init__(self, config: HanaConfig, ctx: PipelineContext):
         super().__init__(config, ctx, "hana")
-
+        
     @classmethod
     def create(cls, config_dict: Dict[str, Any], ctx: PipelineContext) -> "HanaSource":
         config = HanaConfig.parse_obj(config_dict)
         return cls(config, ctx)
+
+    def get_workunits(self) -> Iterable[Union[MetadataWorkUnit, SqlWorkUnit]]: 
+        yield from super().get_workunits()
+
+        if self.config.include_view_lineage: 
+            yield from self._get_view_lineage_workunits()
+
+    def _get_view_lineage_elements(self) -> Dict[Tuple[str, str], List[str]]:
+        """Connects to SAP HANA db to run the query statement. 
+        Returns a dictionary of elements from the query results, which are mapped to the ViewLineageEntry fields"""
+
+        url = self.config.get_sql_alchemy_url() 
+    
+        engine= create_engine(url, **self.config.options)
+
+        data: List[ViewLineageEntry] = [] 
+
+        with engine.connect() as conn: 
+
+            query_results =  conn.execute(LINEAGE_QUERY) 
+    
+            if not query_results.returns_rows: 
+                print("No rows returned.")
+                return {}
+            
+            for row in query_results:
+                data.append(ViewLineageEntry.parse_obj(row))
+        
+        lineage_elements: Dict[Tuple[str, str], List[str]] = defaultdict(list)
+
+        for lineage in data: 
+            if not self.config.view_pattern.allowed(lineage.dependent_view): 
+                self.report.report_dropped(f"{lineage.dependent_schema}.{lineage.dependent_view}")
+                continue
+
+            if not self.config.schema_pattern.allowed(lineage.dependent_view): 
+                self.report.report_dropped(f"{lineage.dependent_schema}.{lineage.dependent_view}")
+                continue
+
+            key = (lineage.dependent_view, lineage.dependent_schema)
+            
+            lineage_elements[key].append(
+                mce_builder.make_dataset_urn(
+                self.platform, 
+                self.config.get_identifier(lineage.source_schema, lineage.source_table), 
+                self.config.env) )
+            
+        return lineage_elements 
+    
+    def _get_view_lineage_workunits(self) -> Iterable[MetadataWorkUnit]: 
+        """Creates MetadataWorkUnit objects for the upstream and downstream lineage of views in the SAP HANA database. 
+        Returns an iterable MetadataWorkUnit that are emitted to Datahub.
+        """
+    
+        lineage_elements = self._get_view_lineage_elements()
+
+        if not lineage_elements: 
+            print("No lineage")
```

Comment:
We can remove this, since we only used it for debugging _or_ use a logger at the debug level